src/CoAPNet.Dtls/Server/CoapDtlsServerTransport.cs

```diff
@@ -196,6 +199,45 @@ private async Task HandleIncoming()
             }
         }
 
+        private byte[]? GetConnectionId(byte[] packet)
+        {
+            try
+            {
+                if (!_connectionIdLength.HasValue)
+                    return null;
+
+                const int headerCidOffset = 11;
+                var cidLength = _connectionIdLength.Value;
+
+                if (packet.Length >= (headerCidOffset + cidLength) && packet[0] == ContentType.tls12_cid)
+                {
+                    var cid = new byte[cidLength];
+                    Array.Copy(packet, headerCidOffset, cid, 0, cidLength);
+                    return cid;
+                }
+
+                return null;
+            }
+            catch (Exception ex)
+            {
```

Comment:
I think logging the packet parameter like this here would just result in "System.Byte[]" in the log right?
If so that seams fairly pointless.

Reply:
It would appear so, but no.
When using Serilog, byte arrays are converted to their hex representation (like `54455354`).

When using the default Microsoft.Extensions.Logging Log Formatter (like when using `Microsoft.Extensions.Logging.Console`) `ToString()` will be called [on each item if the log value is IEnumerable](https://github.com/dotnet/runtime/blob/11fbd151912a3c5d1e72386fb037595b71e214d6/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LogValuesFormatter.cs#L281-L298) (which causes the output to be `84, 69, 83, 84`).

Not great, not terrible but I didn't want to introduce a helper function for this edge case (this is only a fail-safe anyways, if the length-check is correct this should never be hit).

Reply:
Fair enough. Seams reasonable.